src/core/xfa/xhtml.js

```diff
@@ -321,7 +326,16 @@ class XhtmlObject extends XmlObject {
 class A extends XhtmlObject {
   constructor(attributes) {
     super(attributes, "a");
-    this.href = attributes.href || "";
+    let href = "";
+    if (typeof attributes.href === "string") {
+      let url = addDefaultProtocolToUrl(attributes.href);
+      url = tryConvertUrlEncoding(url);
+      const absoluteUrl = createValidAbsoluteUrl(url);
```

Comment:
One thing I'm unsure of is if URLs in XFA documents can ever be relative (since that unfortunately happens sometimes in regular PDF documents)?
If so, we'd need to (somehow) pass along the `pdfManager.docBaseUrl`-parameter when initializing the `XFAFactory`.

Reply:
It'd be weird to have relative URLs... but who know...
@marco-c, could you check this in your corpus ?

Reply:
Here are some of the text found by the `href=\"([^\"]+)\"` regex:
```
'https://www.canada.ca/en/revenue-agency/services/tax/canada-pension-plan-cpp-employment-insurance-ei-rulings/international-social-security-agreements-canada-pension-plan.html': 104, 'http://www.canada.ca/decisions-rpc-ae': 104, 'https://www.canada.ca/fr/agence-revenu/services/impot/decisions-concernant-regime-pensions-canada-rpc-assurance-emploi-ae/accords-internationaux-securite-sociale-regime-pensions-canada.html': 104, 'http://www.canada.ca/cpp-ei-rulings': 94, 'https://www.priv.gc.ca/en/report-a-concern/file-a-formal-privacy-complaint/file-a-complaint-about-a-federal-institution/': 27, 'http://infosource.gc.ca': 12, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications.html': 12, 'https://www.canada.ca/fr/agence-revenu/services/services-electroniques/services-ouverture-session-arc.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_arc-services-ouverture-session': 11, 'https://www.canada.ca/en/revenue-agency/services/e-services/cra-login-services.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_cra-login-services': 11, 'https://www.canada.ca/en/revenue-agency/services/tax/canada-pension-plan-cpp-employment-insurance-ei-rulings.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_cpp-ei-rulings': 11, 'https://www.canada.ca/en/revenue-agency/services/e-services/e-services-individuals/account-individuals.html': 11, 'https://www.canada.ca/fr/immigration-refugies-citoyennete/services/passeports-canadiens.html?utm_source=slash-passport_passeport&amp;utm_medium=short-url-fr&amp;utm_campaign=passport': 10, 'http://www.travel.gc.ca/': 10, '..\\..\\..\\Logos\\!1 Official logos from corporate affairs website\\black\\Canada wordmark\\wordmarkblke.jpg': 9, 'https://www.canada.ca/en/revenue-agency/services/forms-publications.html': 8, 'L:\\FormData\\Graphics\\FIP_Red\\canada_RGB.jpg': 7, 'https://www.canada.ca/fr/services/impots/tpstvh.html': 7, 'http://www.canada.ca/taxes': 6, 'https://www.canada.ca/fr/services/impots.html': 6, 'https://www.canada.ca/en/immigration-refugees-citizenship/corporate/transparency/access-information-privacy/info-source/sources-information.html': 6, 'https://www.canada.ca/en/services/taxes/gsthst.html': 6, 'https://www.canada.ca/fr/agence-revenu/services/services-electroniques/services-electroniques-particuliers/dossier-particuliers.html': 6, 'www.infosource.gc.ca': 6, 'https://www.canada.ca/en/revenue-agency/corporate/about-canada-revenue-agency-cra/access-information-privacy-canada-revenue-agency/info-source-sources-federal-government-employee-information.html': 6, 'https://www.canada.ca/en/services/taxes.html': 5, 'http://www.cic.gc.ca/english/pdf/kits/forms/IMM5476E.pdf': 5, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/medical-police/police-certificates/how.html': 5, 'www.cic.gc.ca': 5, 'https://www.canada.ca/en/revenue-agency/services/tax/businesses/topics/gst-hst-businesses/forms-publications.html': 5, 'http://www.canada.ca/impot': 4, 'https://www.canada.ca/en/revenue-agency/corporate/about-canada-revenue-agency-cra/access-information-privacy-canada-revenue-agency/info-source-sources-federal-government-employee-information.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_cra-info-source': 4, 'http://www.cra-arc.gc.ca/myaccount/': 4, 'http://www.cic.gc.ca/english/pdf/kits/forms/IMM5475E.pdf': 4, 'http://www.cic.gc.ca/english/pdf/kits/forms/IMM0008ENU_2D.pdf': 4, 'http://www.cic.gc.ca/english/pdf/kits/forms/eIMM5669E.pdf': 4, 'http://www.cic.gc.ca/english/information/applications/spouse.asp#country': 4, 'https://www.canada.ca/content/dam/ircc/migration/ircc/english/information/applications/guides/pdf/pc_lettera_e.pdf': 4, 'http://www.cic.gc.ca/english/pdf/kits/forms/IMM1344E.pdf': 4, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5482-instruction-fill-financial-evaluation-form-1283.html': 4, 'http://www.cic.gc.ca/english/pdf/kits/forms/IMM5406E.pdf': 4, 'http://www.cic.gc.ca/english/information/applications/photospecs.asp': 4, 'http://www.cic.gc.ca/english/visit/biometrics.asp': 4, 'https://www.canada.ca/en/revenue-agency/services/e-services/cra-mobile-apps.html': 4, 'https://www.canada.ca/en/revenue-agency/services/tax/individuals/topics/tax-free-savings-account.html': 4, 'http://www.canada.ca/arc-coronavirus': 4, 'https://www.canada.ca/en/revenue-agency/services/e-services/e-services-businesses/business-account.html': 4, 'http://www.cra-arc.gc.ca/formspubs/menu-fra.html': 4, 'https://www.canada.ca/fr/agence-revenu/services/impot/particuliers/sujets/compte-epargne-libre-impot.html': 4, 'mailto:ATIP-AIPRP@cra-arc.gc.ca': 4, 'mailto:aafc.atip-aiprp.aac@canada.ca': 4, 'http://Infosource.gc.ca': 3, 'https://travel.gc.ca/': 3, 'https://www.canada.ca/en/revenue-agency/services/tax/individuals/topics/rrsps-related-plans/what-home-buyers-plan.html': 3, 'https://www.canada.ca/fr/agence-revenu/services/impot/particuliers/sujets/reer-regimes-connexes/est-regime-accession-a-propriete.html': 3, 'http://www.canada.ca/arc-info-source': 3, 'www.arc.gc.ca/formulaires': 3, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/medical-police/police-certificates/about.html': 3, 'https://www.canada.ca/en/immigration-refugees-citizenship/news/video/ use-representative-form-5476.html': 3, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/tps-tvh-entreprises/formulaires-publications-lies-a-tps-tvh.html': 3, 'https://voyage.gc.ca/': 3, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/tps-tvh-entreprises/formulaires-publications.html': 3, 'https://www.canada.ca/fr/agence-revenu/services/services-electroniques/services-electroniques-entreprises/dossier-entreprise.html': 3, 'https://www.canada.ca/en/revenue-agency/services/tax/businesses/topics/gst-hst-businesses/gst-hst-related-forms-publications.html': 3, 'https://www.canada.ca/fr/agence-revenu/organisation/a-propos-agence-revenu-canada-arc/acces-a-information-protection-renseignements-personnels-a-agence-revenu-canada/info-source-sources-renseignements-gouvernement-federal-fonctionnaires-federaux.html': 3, 'http://canada.ca/impots-incitatif-investissement-accelere': 3, 'http://infosource.gc.ca/': 3, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/': 3, 'http://www.cic.gc.ca/english/information/pr-card/apply-photos.asp': 3, 'http://www.cic.gc.ca/english/pdf/kits/forms/IMM5532E.pdf': 3, 'http://www.cic.gc.ca/english/pdf/kits/forms/IMM5604E.pdf': 3, 'http://www.cic.gc.ca/english/information/representative/non-rep.asp': 3, 'http://www.cic.gc.ca/english/pdf/kits/forms/IMM0008DEPENU.pdf': 3, 'http://www.cic.gc.ca/english/information/applications/spouse.asp': 3, 'http://canada.ca/impots': 3, 'http://www23.statcan.gc.ca/imdb/p3VD.pl?Function=getVDPage1&amp;db=imdb&amp;dis=2&amp;adm=8&amp;TVD=118464': 3, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/citizenship-application-photograph-specifications.html': 3, 'https://www.canada.ca/content/dam/ircc/migration/ircc/english/pdf/kits/citizen/cit0404e-2.pdf': 3, 'https://www.canada.ca/content/dam/ircc/migration/ircc/english/pdf/kits/forms/imm5476e.pdf': 3, 'https://eservices.cic.gc.ca/epay/welcome.do?lang=en&amp;_ga=1.83255318.1488559817.1428960024': 3, 'https://www.canada.ca/content/dam/ircc/migration/ircc/english/pdf/kits/citizen/cit0464e.pdf': 3, 'https://www.canada.ca/fr/agence-revenu/services/services-electroniques/applications-mobiles-arc.html': 3, 'http://get.adobe.com/uk/reader/': 2, 'http://www.gpo.gov/fdsys/pkg/FR-2011-10-28/html/2011-27881.htm': 2, 'http://www.cra-arc.gc.ca/F/pub/tg/t4061/t4061-f.html': 2, 'http://www23.statcan.gc.ca:81/imdb/p3VD_f.pl?Function=getVDPage1&amp;db=imdb&amp;dis=2&amp;adm=8&amp;TVD=118464': 2, 'http://www.cra-arc.gc.ca/E/pub/tg/t4061/t4061-e.html': 2, 'servicecanada.gc.ca': 2, 'http://www.cra-arc.gc.ca/gncy/tp/nfsrc/nfsrc-eng.html': 2, 'arc.gc.ca/formulaires': 2, 'https://www.canada.ca/fr/gouvernement/ouvrir-session-dossier-compte-en-ligne.html': 2, 'https://canada.ca/cra-info-source': 2, 'http://www.cic.gc.ca/english/information/applications/guides/pdf/5445EB-e.pdf': 2, 'https://www.priv.gc.ca/fr/signaler-un-probleme/deposer-une-plainte-officielle-concernant-la-protection-de-la-vie-privee/deposer-une-plainte-visant-une-institution-du-gouvernement-federal/': 2, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5289-sponsor-your-spouse-common-law-partner-conjugal-partner-dependent-child-complete-guide.html#0008': 2, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/immigration-citizenship-representative.html': 2, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5289-sponsor-your-spouse-common-law-partner-conjugal-partner-dependent-child-complete-guide.html#1344': 2, 'http://laws-lois.justice.gc.ca/eng/regulations/sor-2002-227/page-10.html#h-25': 2, 'https://www.canada.ca/en/immigration-refugees-citizenship/corporate/transparency/access-information-privacy/info-source/personal-information-banks.html': 2, 'https://www.canada.ca/fr/agence-revenu/services/autochtones.html': 2, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-sponsor-refugee-community.html': 2, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications/publications/t4061.html': 2, 'https://www.canada.ca/fr/agence-revenu/services/impot/impot-international-non-residents/accords-avis-autorite-competente/formulaire-t106-declaration-renseignements-operations-lien-dependance-effectuees-non-residents-politique-minimis.html': 2, 'http://www23.statcan.gc.ca/imdb/p3VD_f.pl?Function=getVDPage1&amp;db=imdb&amp;dis=2&amp;adm=8&amp;TVD=118464': 2, 'http://www.fin.gc.ca': 2, 'https://canada.ca/en/immigration-refugees-citizenship/corporate/mandate/policies-operational-instructions-agreements/permanent-residence-healthcare-pandemic-canada.html': 2, 'https://www.canada.ca/en/revenue-agency/corporate/contact-information/tax-services-offices-tax-centres.html': 2, 'http://www.canada.ca/cra-coronavirus': 2, 'https://www.canada.ca/en/revenue-agency/services/e-services/represent-a-client.html': 2, 'mailto:VSCQuestions.EnquetesCSAC@forces.gc.ca': 2, 'https://www.canada.ca/en/revenue-agency/services/e-services/payment-save-time-pay-online.html': 2, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5525-basic-guide-sponsor-your-spousepartner-': 2, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/applicationforms-guides/guide-5289-sponsor-your-spouse-common-law-partner-conjugal-partner-dependent-child-completeguide.html#5532': 2, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-formsguides/guide-5289-sponsor-your-spouse-common-law-partner-conjugal-partner-dependent-child-complete-guide.html#0008': 2, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/application-citizenshipcertificate-adults-minors.html': 2, 'http://laws-lois.justice.gc.ca/eng/regulations/sor-2002-227/page-10.html': 2, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5289-sponsor-your-spouse-common-law-partner-conjugal-partnerdependent-child-complete-guide.html': 2, 'https://www.canada.ca/fr/agence-revenu/organisation/a-propos-agence-revenu-canada-arc/acces-a-information-protection-renseignements-personnels-a-agence-revenu-canada/info-source-sources-renseignements-gouvernement-federal-fonctionnaires-federaux.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_arc-info-source': 2, 'https://www.canada.ca/content/dam/ircc/migration/ircc/english/pdf/kits/citizen/cit0407e-2.pdf': 2, 'https://www.canada.ca/content/dam/ircc/migration/ircc/english/pdf/kits/citizen/cit0177e-2.pdf': 2, 'https://www.canada.ca/en/revenue-agency/services/tax/international-non-residents/competent-authority-agreements-notices/form-t106-information-return-non-arms-length-transactions-non-residents-minimis-policy.html': 2, 'https://www.canada.ca/en/revenue-agency/services/forms-publications/publications/t4061.html': 2, 'http://www.fin.gc.ca/': 2, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-sponsor-refugee-groups-five.html': 2, 'www.arc.gc.ca/gncy/tp/nfsrc/nfsrc-fra.html': 2, 'http://www.canada.ca/arc-dispositions-non-residents': 2, '../pdf/quickstarts/rbm signer.pdf': 1, 'mailto:SANCO-BO@ec.europa.eu': 1, 'mailto:SANCO-VET-PROG@ec.europa.eu?subject=Veterinary%20programmes%20-%20issue': 1, 'http:': 1, 'http://www.gmc-uk.org/guidance/ethical_guidance/raising_concerns.asp': 1, '..\\images\\NHS_Help_crop.gif': 1, 'http://www.legislation.gov.uk/ukpga/1998/29/introduction/enacted': 1, 'http://www.legislation.gov.uk/ukpga/1998/29/contents': 1, 'www.legislation.gov.uk/id/ukdsi/2010/9780111500286': 1, 'http://www.gmc-uk.org/index.asp': 1, 'http://www.dh.gov.uk/en/Publicationsandstatistics/Publications/PublicationsPolicyAndGuidance/DH_4069253': 1, 'https://www.gov.uk/government/uploads/system/uploads/attachment_data/file/192572/2900774_InfoGovernance_accv2.pdf': 1, 'http://www.revalidationsupport.nhs.uk/about_the_rst/rst_projects/Implementation-Support/guidance-and-tools/about_Information_Guidance.php': 1, 'http://www.gmc-uk.org/doctors/revalidation/revalidation_information.asp': 1, 'https://www.gov.uk/government/uploads/system/uploads/attachment_data/file/200146/Confidentiality_-_NHS_Code_of_Practice.pdf': 1, 'http://www.legislation.gov.uk/uksi/2013/391/made': 1, 'http://www.nhsemployers.org/Aboutus/Publications/Pages/Guiding-principles-for-sharing-information-on-healthcare-workers.aspx': 1, 'https://www.gamblingcommission.gov.uk/for-gambling-businesses/Apply-for-a-licence/Annual-fees.aspx': 1, 'http://www.michigan.gov/documents/deq/deq-rrd-officemap-EQP4410_247028_7.pdf': 1, 'www.hmrc.gov.uk/payerti/payroll/pay-and-deductions/student-loan.htm': 1, '\\\\Irf03276\\cam-cit shared general\\Delivery\\Technical team\\iPDF work in progress\\Component Library\\Logos\\HMRC06_RGB_BLK_HR_45mm.jpg': 1, 'www.hmrc.gov.uk/payerti/employee-starting/index.htm': 1, '..\\images\\logo-inps.jpg': 1, 'mailto:FiscalServices@dps.state.nv.us': 1, '.#som($template.#subform.designer__stylesheet.LessMargin)': 1, '.#som($template.#subform.designer__stylesheet.SubformStyle2)': 1, '..\\..\\..\\..\\ADOBE_MODULISTICA_VECCHIA\\GesPiaAdo_CVS\\GesPiaAdoEtc\\adobe\\resources\\Modulistica Vecchia\\Frammenti\\images\\logo_inps_inpdap.jpg': 1, 'http://www.hmrc.gov.uk/inheritancetax/intro/basics.htm': 1, 'http://www.hmrc.gov.uk/taxcredits/start/claiming/get-started/how-to-claim.htm': 1, 'http://www.hmrc.gov.uk/record-keeping/index.htm': 1, 'http://www.hmrc.gov.uk/campaigns/emarket-guide.pdf': 1, 'www.hmrc.gov.uk': 1, 'http://www.businesslink.gov.uk/bdotg/action/home?domain=www.businesslink.gov.uk&amp;target=http://www.businesslink.gov.uk/': 1, 'http://www.hmrc.gov.uk/taxcredits/start/who-qualifies/quick-questionnaire.htm': 1, 'mailto:ris.campaigns@hmrc.gsi.gov.uk': 1, 'www.hmrc.gov.uk/online/new.htm': 1, 'www.hmrc.gov.uk/working/intro/class2.htm': 1, 'www.hmrc.gov.uk/cgt/intro/basics.htm': 1, 'http://www.hmrc.gov.uk/rates/it.htm': 1, 'http://www.hmrc.gov.uk/working/intro/class2.htm': 1, 'http://www.hmrc.gov.uk/incometax/relief-self-emp.htm': 1, 'http://www.hmrc.gov.uk/vat/start/register/when-to-register.htm': 1, 'www.hmrc.gov.uk/campaigns/5years-calc.pdf': 1, 'http://www.haw-hamburg.de/fakultaeten-und-departments/ls/services/fsb-ls.html': 1, 'https://www.haw-hamburg.de/fakultaeten-und-departments/ls/fakultaetsservicebuero.html': 1, '..\\..\\..\\HAW Logo\\HAW Logo NEU.gif': 1, 'https://www.haw-hamburg.de/hochschule/life-sciences/studium-und-lehre/fakultaetsservicebuero/abschlussarbeiten-und-zeugnisdokumente/': 1, 'https://secure.ssa.gov/apps6z/FOLO/fo001.jsp': 1, 'www.ssa.gov/privacy': 1, 'www.socialsecurity.gov': 1, 'mailto:hrservicecenter@worldbank.org?subject=Request%20for%20PID%20Number%20Submission': 1, 'http://www.worldbank.org/en/about/unit/human-resources/request-g5-visa-for-a-prospective-domestic-employee-residing-outside-the-us?cq_ck=1553523949254': 1, 'http://hrforms/': 1, 'http://myhrss.worldbank.org': 1, 'https://www.uscg.mil/Resources/legal/DRB/': 1, 'http://dpcld.defense.gov/Privacy/SORNsIndex/DOD-Component-Article-View/Article/569821/f036-af-pc-c/': 1, 'http://www.afpc.af.mil/Separation/Discharge-Review-Board/': 1, 'http://www.nara.gov/regional/mpr.html': 1, 'http://www.secnav.navy.mil/mra/CORB/pages/ndrb/default.aspx': 1, 'http://arba.army.pentagon.mil': 1, 'http://en.wikipedia.org/wiki/Tank_Wars': 1, 'http://forms.stefcameron.com/xfa-tank-wars/': 1, 'http://www.cra-arc.gc.ca/tx/nnrsdnts/cmmn/frgn/menu-fra.html': 1, 'http://www23.statcan.gc.ca:81/imdb/p3VD.pl?Function=getVDPage1&amp;db=imdb&amp;dis=2&amp;adm=8&amp;TVD=118464': 1, 'http://www.cra.gc.ca/E/pub/tg/t4061/t4061-e.html': 1, 'http://www.Canada.ca/passport': 1, 'http://gcdocs2/otcsdav/nodes/384244698/phac.privacy-vieprivee.aspc@canada.ca': 1, 'http://www.irb-cisr.gc.ca/eng/port/refasi/cladem/pages/index.aspx': 1, 'http//www.cic.gc.ca/english/refugees/inside/index.asp': 1, 'https://www.canada.ca/en/employment-social-development/services/sin/apply.html': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications/forms/t1159.html': 1, 'www.cra.gc.ca/forms': 1, 'www.cra.gc.ca': 1, 'https://laws-lois.justice.gc.ca/eng/acts/E-5.6/': 1, 'http://laws-lois.justice.gc.ca/eng/acts/O-9/': 1, 'http://noc.esdc.gc.ca/English/noc/welcome.aspx?ver=16': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/businesses/topics/gst-hst-businesses/gst-hst-rebates/foreign-convention-tour.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/tps-tvh-entreprises/remboursements-tps-tvh/etrangers-voyages-organises.html': 1, 'http://www.cic.gc.ca/english/information/applications/photospecs-cit.asp': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/decisions-concernant-regime-pensions-canada-rpc-assurance-emploi-ae/a-propos-regime-pensions-canada-assurance-emploi/sectes-religieuses-regime-pensions-canada.html': 1, 'cra.gc.ca/representatives': 1, 'cra.gc.ca/mybusinessaccount': 1, 'cra-arc.gc.ca/gncy/tp/nfsrc/nfsrc-eng.html': 1, 'http://www.canada.ca/impots': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/decisions-concernant-regime-pensions-canada-rpc-assurance-emploi-ae/avez-vous-recu-decision-relative-rpc-a-e.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/decisions-concernant-regime-pensions-canada-rpc-assurance-emploi-ae/a-propos-regime-pensions-canada-assurance-emploi/succession-employeurs-employeur-faisant-objet-restructuration.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications/publications/rc4110.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/decisions-concernant-regime-pensions-canada-rpc-assurance-emploi-ae/a-propos-regime-pensions-canada-assurance-emploi/lien-dependance-fins-loi-assurance-emploi.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/decisions-concernant-regime-pensions-canada-rpc-assurance-emploi-ae.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_decisions-rpc-ae': 1, 'http://www.cic.gc.ca/english/pdf/kits/forms/IMM1283E.pdf': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5525-basic-guide-sponsor-your-spouse-partner-child.html#fees': 1, 'http:// https://www.canada.ca/en/immigration-refugees-citizenship/services/new-immigrants/pr-card/apply-renew-replace/photo.html': 1, 'http://www.cic.gc.ca/english/pdf/kits/forms/IMM5481E.pdf': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5289-sponsor-your-spouse-common-law-partner-conjugal-partner-dependent-child-complete-guide.html#5406': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/immigration-citizenship-representative/application-access.html': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5289-sponsor-your-spouse-common-law-partner-conjugal-partner-dependent-child-complete-guide.html#5481': 1, 'http://www.cic.gc.ca/english/information/applications/guides/5759ETOC.asp': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/individuals/topics/about-your-tax-return/making-payments-individuals/paying-your-income-tax-instalments.html': 1, 'http://www.irb-cisr.gc.ca/eng/tribunal/form/Documents/boc_fda_e.pdf': 1, 'http://canada.ca/taxes-travel-costs': 1, 'http://canada.ca/taxes': 1, 'http://www.iso.org/iso/fr/home/standards/country_codes.htm?=': 1, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications/publications/rc4651.html': 1, 'http://www.banqueducanada.ca/taux/taux-de-change/': 1, '..\\..\\..\\Graphics\\ATIP\\signature.PNG': 1, '..\\..\\..\\Graphics\\ATIP\\scanner.PNG': 1, '..\\..\\..\\Graphics\\ATIP\\Family.PNG': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/tps-tvh-entreprises.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/particuliers/sujets/tout-votre-declaration-revenus/comment-modifier-votre-declaration.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_modifier-declaration-revenus': 1, 'https://www.canada.ca/fr/agence-revenu/services/services-electroniques/services-electroniques-entreprises/retransmettre-demande-redressement-ligne-declaration-t1-fournisseurs-service.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_retransmettre': 1, 'https://www.canada.ca/fr/agence-revenu/services/services-electroniques/services-electroniques-particuliers/dossier-particuliers.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_mon-dossier-arc': 1, 'http://www.cic.gc.ca/english/information/applications/guides/5568ETOC.asp#appendixA': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/renseignements-techniques/impot-revenu/folios-impot-revenu/serie-4-entreprises/serie-4-entreprises-folio-16-societe-personnes/folio-impot-revenu-s4-f16-c1-est-societe-personnes.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/revenu-petites-entreprises-travailleurs-independants/etablir-votre-entreprise/apport-biens-entreprise/juste-valeur-marchande.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/administrateurs-fiducies/declaration-t3/residence-comment-nous-joindre.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/societes/credits-dimpot-entreprises/organisation-journalistique-canadienne-admissible.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/revenu-petites-entreprises-travailleurs-independants/etablir-votre-entreprise.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/societes/credits-dimpot-entreprises/credit-impot-main-oeuvre-journalistique-canadienne/lignes-directrices.html#nouvellesoriginales': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/societes/credits-dimpot-entreprises/credit-impot-main-oeuvre-journalistique-canadienne/lignes-directrices.html#emploi': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/renseignements-techniques/impot-revenu/folios-impot-revenu/serie-1-particuliers/serie-1-particuliers-folio-5-transfert-revenu-biens-droits-a-tiers/folio-impot-revenu-s1-f5-c1-personnes-liees-personnes-sans-lien-dependance-entre-elles.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/renseignements-techniques/impot-revenu/folios-impot-revenu/serie-6-fiducies/serie-6-fiducies-folio-1-questions-liees-fiducies-a-residence/folio-impot-revenu-s6-f1-c1-residence-fiducie-succession.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/societes/credits-dimpot-entreprises/credit-impot-main-oeuvre-journalistique-canadienne/lignes-directrices.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications/formulaires/aut-01.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications/formulaires/t625.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/impot-international-non-residents/entreprises-impot-international-non-residents/statut-residence-societe.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/particuliers/sujets/tout-votre-declaration-revenus/declaration-revenus/remplir-declaration-revenus/deductions-credits-depenses/toutes-deductions-tous-credits-toutes-depenses.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/autochtones/renseignements-indiens.html#hdng4': 1, 'http://www.cic.gc.ca/english/department/atip/infosource/pibs.asp': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/businesses/topics/sole-proprietorships-partnerships/report-business-income-expenses/claiming-capital-cost-allowance/accelerated-investment-incentive.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/a-propos-agence-revenu-canada-arc/plaintes-differends/faire-plainte-liee-service.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/autorisation-representant.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/services-electroniques/representer-client.html': 1, 'www.cra.gc.ca/tso': 1, 'http://www.cic.gc.ca/english/department/atip/form-imm5744.asp': 1, 'http://www.cic.gc.ca/english/department/atip/index.asp': 1, 'http://www.cic.gc.ca/english/information/applications/certcopy.asp': 1, 'http://www.cic.gc.ca/english/information/applications/guides/5561E.asp': 1, 'mailto:name@provider.net': 1, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications/formulaires/t106.html': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/immigrate-canada/start-visa/eligibility.html#money': 1, 'http://www.immigration-quebec.gouv.qc.ca/en/informations/news/news-2020/asylum-seekers-health.html': 1, 'https://www.canada.ca/content/dam/ircc/migration/ircc/english/information/applications/guides/pdf/5445eb-e.pdf': 1, 'https://www.canada.ca/en/revenue-agency/programs/about-canada-revenue-agency-cra/acts-regulations/list-acts.html': 1, 'https://w&#xD;ww.priv.gc.ca/en/report-a-concern/file-a-formal-privacy-complaint/file-a-complaint-about-a-federal-institution/': 1, 'http://canada.ca/taxes-accelerated-investment-income': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/particuliers/sujets/regime-pension-agree-collectif-rpac-information-particuliers.html': 1, 'https://www.canada.ca/en/revenue-agency/services/about-canada-revenue-agency-cra/complaints-disputes/make-a-service-complaint.html': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/representative-authorization.html': 1, 'https://www.canada.ca/en/services/taxes/savings-and-pension-plans/savings-and-pension-plan-administration.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_registered-plans-administrators': 1, 'mailto:IRCC.FINIIP-ENFeeReturns-RemboursementsdesfraisPII-PEFIN.IRCC@cic.gc.ca?subject=Return%20of%20Application%20Fees%20for%20the%20Federal%20Immigrant%20Investor%20Program%20(IIP)%20and%20Entrepreneur%20Program%20(EN)': 1, 'http://www.cra-arc.gc.ca/forms/': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications/publications/gi-044/tour-packages-what-eligible-tour-package.html': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/businesses/topics/gst-hst-businesses/gst-hst-rebates/foreign-convention-tour-incentive-program.html': 1, '..\\..\\..\\..\\Logos\\!1 Official logos from corporate affairs website\\black\\Canada wordmark\\wordmarkblke.jpg': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_cra-forms-publications': 1, 'https://laws-lois.justice.gc.ca/fra/lois/P-21/page-1.html': 1, 'https://www.canada.ca/fr/ministere-defense-nationale/organisation/transparence/acces-information-protection-renseignements-personnels/info-source-sources-de-renseignements-du-gouvernement-federal-et-sur-les-fonctionnaires-federaux.html': 1, 'https://www.canada.ca/en/revenue-agency/services/make-a-payment-canada-revenue-agency.html': 1, 'https://www.canada.ca/en/revenue-agency/corporate/about-canada-revenue-agency-cra/access-information-privacy-canada-revenue-agency/info-source-sources-federal-government-employee-information.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_cra-info-source#txsrvcsndprcssng': 1, 'http://www.cic.gc.ca/english/information/applications/guides/5445ETOC.asp': 1, 'phac.privacy-vieprivee.aspc@canada.ca': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5525-basic-guide-sponsor-your-spouse-partner-child.html#conjugal': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_cra-forms': 1, 'http://www.canada.ca/mon-dossier-arc': 1, 'http://www.canada.ca/arc-applications-mobiles': 1, 'http://www.Canada.ca/passeport': 1, 'www.cra.gc.ca/gncy/tp/nfsrc/nfsrc-eng.html': 1, 'http://canada.ca/impots-frais-deplacement': 1, 'https://www.canada.ca/cra-forms-publications': 1, 'http://canada.ca/my-cra-account': 1, 'http://canada.ca/cra-represent-a-client': 1, 'http://www.canada.ca.Represent a Client': 1, 'https://canada.ca/taxes-authorize-representative': 1, 'https://canada.ca/taxpayer-relief': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications/forms/rc376.html': 1, 'https://www.canada.ca/fr/services/impots/regimes-depargne-et-de-pension/administration-des-regimes-depargne-et-de-pension.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_administrateurs-regimes-enregistres': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#ListItems': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#TableSummary': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#HiddenAnnot': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#NavLinks': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#LogicalRO': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#Perms': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#FigAltText': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#LblLBody': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#CharEnc': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#PrimeLang': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#OtherAltText': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#TaggedFormFields': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#Multimedia': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#Bookmarks': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#NestedAltText': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#TabOrder': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#TaggedPDF': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#Scripts': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#ColorContrast': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#Headings': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#FormFieldNames': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#THTD': 1, '#reportTop': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#TimedResponses': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#TaggedAnnots': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#TableHeaders': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#TaggedCont': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#TableRows': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#FlickerRate': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#DocTitle': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#ImageOnlyPDF': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#AltTextNoContent': 1, 'http://www.adobe.com/go/acrobat11_accessibility_checker_en#RegularTable': 1, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications/publications/gi-044/voyages-organises-est-voyage-organise-admissible.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/tps-tvh-entreprises/remboursements-tps-tvh/programme-incitation-congres-etrangers-voyages-organises.html': 1, 'http://www.cra-arc.gc.ca/tx/nnrsdnts/cmmn/frgn/menu-eng.html': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0403-citizenship-minors-subsection-5-1.html#Appendix_C': 1, 'https://eservices.cic.gc.ca/rescalc/resCalcStartNew.do': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0403-citizenship-minors-subsection-5-1.html#Appendix_A': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0403-citizenship-minors-subsection-5-1.html#Appendix_B': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0403-citizenship-minors-subsection-5-1.html': 1, 'https://www.canada.ca/content/dam/ircc/migration/ircc/english/pdf/kits/citizen/cit0403e.pdf': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications/forms/t106.html': 1, 'mailto:IRCC.FINFSWFeeReturns-RemboursementsdesfraisTQFFIN.IRCC@cic.gc.ca?subject=Return%20of%20Application%20Fees%20': 1, 'http://www.cra-arc.gc.ca/esrvc-srvce/tx/ndvdls/myccnt/menu-fra.html': 1, 'http://www.cra-arc.gc.ca/gncy/tp/nfsrc/nfsrc-fra.html': 1, 'https://eservices.cic.gc.ca/rescalc/resCalcStartNew.do?lang=eng': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0002-application-canadian-citizenship-under-subsection-5-1-adults-18-years-older.html#AppendixC': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0002-application-canadian-citizenship-under-subsection-5-1-adults-18-years-older.html#AppendixA': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0002-application-canadian-citizenship-under-subsection-5-1-adults-18-years-older.html#AppendixB': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0002-application-canadian-citizenship-under-subsection-5-1-adults-18-years-older.html': 1, 'https://www.canada.ca/content/dam/ircc/documents/pdf/english/kits/forms/cit0002e-2.pdf': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/corporate/transparency/access-information-privacy/info-source/personal-information-banks.html#migrationcontrol': 1, 'https://www.canada.ca/en/revenue-agency/services/e-services/e-services-businesses/business-account.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_my-cra-business-account': 1, 'https://www.canada.ca/en/revenue-agency/services/e-services/represent-a-client.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_taxes-representatives': 1, 'https://www.canada.ca/impots-incitatif-investissement-accelere': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/application-citizenship-certificate-adults-minors.html': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/news/video/use-representative-form-5476.html': 1, 'http://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5289-sponsor-your-spouse-common-law-partner-conjugal-partner-dependent-child-complete-guide.html#5406': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5289-sponsor-your-spouse-common-law-partner-conjugal-partner-dependent-child-complete-guide.html#5532': 1, 'https://www.canada.ca/content/dam/ircc/migration/ircc/english/pdf/kits/forms/imm1283e.pdf': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5525-basic-guide-sponsor-your-spouse-partner-child.html': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5561-instructions-use-representative.html': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-5289-sponsor-your-spouse-common-law-partner-conjugal-partner-dependent-child-complete-guide.html#5669': 1, 'http://www.canada.ca/child-family-benefits-calculator': 1, 'http://www.canada.ca/cra-forms': 1, 'http://www.canada.ca/cra-info-source': 1, 'www.fin.gc.ca/treaties-conventions/in_force--eng.asp': 1, 'http://www.canada.ca/en/services/benefits/publicpensions/cpp/cpp-international/apply.html': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/international-non-residents/information-been-moved/rendering-services-canada.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_cra-rendering-services-Canada': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-sponsor-refugee-agreement-holder-constituent-group.html': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/businesses/topics/gst-hst-businesses.html': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/individuals/segments/tax-credits-deductions-persons-disabilities/disability-tax-credit.html': 1, 'www.canada.ca/arc-info-source': 1, 'https://www.priv.gc.ca/fr/signaler-u&#xD;n-probleme/deposer-une-plainte-officielle-concernant-la-protection-de-la-vie-privee/deposer-une-plainte-visant-une-institution-du-gouvernement-federal/': 1, 'http://www.cra-arc.gc.ca/tx/ndvdls/tpcs/ncm-tx/rtrn/cmpltng/rprtng-ncm/lns101-170/127/menu-eng.html': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/individuals/topics/about-your-tax-return/tax-return/completing-a-tax-return/deductions-credits-expenses/line-25500-northern-residents-deductions.html': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/individuals/topics/about-your-tax-return/tax-return/completing-a-tax-return/deductions-credits-expenses/line-25500-northern-residents-deductions/meal-vehicle-rates-used-calculate-travel-expenses.html': 1, 'http://www.iso.org/iso/country_codes': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications/publications/rc4651.html': 1, 'http://www.bankofcanada.ca/rates/exchange/': 1, 'https://www.canada.ca/fr/agence-revenu/organisation/a-propos-agence-revenu-canada-arc/acces-a-information-protection-renseignements-personnels-a-agence-revenu-canada/info-source-sources-renseignements-gouvernement-federal-fonctionnaires-federaux.html#srvcsfscxttrtmnt': 1, 'https://www.canada.ca/fr/agence-revenu/services/services-electroniques/paiement-gagnez-temps-payez-ligne.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_mon-paiement-arc': 1, '..\\..\\..\\..\\Graphics\\FIP_Red\\canada_RGB.jpg': 1, 'http://www.osfi-bsif.gc.ca/swppws/default.html': 1, 'https://www.canada.ca/fr/services/impots/regimes-depargne-et-de-pension/administration-des-regimes-depargne-et-de-pension.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_administrateurs-regimes-enregistres#wb-cont': 1, 'https://www.canada.ca/en/revenue-agency/services/e-services/e-services-businesses/gst-hst-netfile.html': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/businesses/topics/gst-hst-businesses/charge-collect-which-rate/financial-institutions.html': 1, 'http://www.canada.ca/taxes-accelerated-investment-income': 1, 'file:///https:\\hich\\af1\\dbch\\af1\\loch\\f1 &#xD;//www.priv.gc.ca/en/report-a-concern/file-a-formal-privacy-complaint/file-a-complaint-about-a-federal-institution/': 1, 'http://canada.ca/cra-info-source': 1, 'http://canada.ca/payments': 1, 'https://www.canada.ca/fr/ministere-defense-nationale/campagnes/covid-19/travailler-distance.html': 1, 'https://www.btb.termiumplus.gc.ca/tpv2alpha/alpha-fra.html?lang=fra&amp;i=1&amp;index=alt&amp;codom2nd_wet=1': 1, 'https://www.canada.ca/fr/ministere-defense-nationale/campagnes/covid-19/travailler-distance/instructions-remplir-atfc.html': 1, 'https://www.tbs-sct.gc.ca/agreements-conventions/index-fra.aspx': 1, 'https://www.canada.ca/fr/agence-revenu/services/services-electroniques/paiement-gagnez-temps-payez-ligne.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/entreprise-individuelle-societe-personnes/declarer-vos-revenus-depenses-entreprise/reclamer-deduction-amortissement/incitatif-investissement-accelere.html': 1, 'https://www.canada.ca/fr/emploi-developpement-social/services/numero-assurance-sociale/demande.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications/formulaires/t1159.html': 1, 'https://www.canada.ca/cra-info-source': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/canada-pension-plan-cpp-employment-insurance-ei-rulings/cpp-ei-explained/meaning-dealing-arms-length-purposes-employment-insurance-act.html': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/canada-pension-plan-cpp-employment-insurance-ei-rulings/have-you-received-a-cpp-ei-ruling.html': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/canada-pension-plan-cpp-employment-insurance-ei-rulings/cpp-ei-explained/employer-restructuring-succession-employers.html': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications/publications/rc4110.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/faire-paiement-a-agence-revenu-canada.html': 1, 'http://canada.ca/arc-info-source': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/businesses/topics/gst-hst-businesses/specific-sectors/gst-hst-financial-institutions-including-selected-listed-financial-institutions.html': 1, 'http://www.cic.gc.ca/francais/ressources/outils/temp/travail/avis/provinces-territoires.asp': 1, 'http://www.cic.gc.ca/francais/visiter/biometrie.asp': 1, 'http://www.cic.gc.ca/francais/information/demandes/photospecs.asp': 1, 'https://www.canada.ca/en/revenue-agency/services/e-services/e-services-individuals/account-individuals.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_my-cra-account': 1, 'https://www.canada.ca/en/revenue-agency/services/e-services/e-services-businesses/refile-online-t1-adjustments-efile-service-providers.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_refile': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/individuals/topics/about-your-tax-return/change-your-return.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_change-tax-return': 1, 'http://www.canada.ca/allegement-contribuables': 1, 'http://www.cic.gc.ca/english/resources/tools/temp/work/opinion/territories-provinces.asp': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/tps-tvh-entreprises/facturer-percevoir-quel-taux/institutions-financieres.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/services-electroniques/services-electroniques-entreprises/impotnet-tps-tvh.html': 1, 'https://laws-lois.justice.gc.ca/eng/acts/C-44.6/section-sched108197.html': 1, 'https://laws-lois.justice.gc.ca/eng/acts/C-44.6/section-sched108325.html': 1, 'http://www.elections.ca': 1, 'https://www.canada.ca/en/revenue-agency/services/about-canada-revenue-agency-cra/direct-deposit.html': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications/forms/rc66-1.html': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications/forms/rc66sch.html': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications/forms/ctb9.html': 1, 'https://www.canada.ca/en/revenue-agency/services/child-family-benefits.html': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/corporate/transparency/access-information-privacy/info-source/personal-information-banks.html#healthprotection': 1, '}{\\rtlch\\fcs1 \\af1\\afs20 \\ltrch\\fcs0 \\b\\f1\\fs20\\lang2057\\langfe1033\\langnp2057\\insrsid5916922\\charrsid5533457 &#xD;http://healthycanadians.gc.ca/diseases-conditions-maladies-affections/disease-maladie/syphilis-eng.php}{\\rtlch\\fcs1 \\af1\\afs20 \\ltrch\\fcs0 \\b\\f1\\fs20\\lang2057\\langfe1033\\langnp2057\\insrsid5916922 ': 1, 'http://www.canada.ca/cra-mobile-apps': 1, 'http://www.canada.ca/my-cra-account': 1, 'www.canada.ca/gst-hst': 1, 'www.canada.ca/taxes-representatives': 1, 'www.canada.ca/gst-hst-netfile': 1, 'www.canada.ca/my-cra-business-account': 1, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications/formulaires/rc66sch-statut-canada-etat-revenus.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/prestations-enfants-familles.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications/formulaires/rc66-1.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/a-propos-agence-revenu-canada-arc/depot-direct.html': 1, 'https://www.canada.ca/fr/agence-revenu/services/formulaires-publications/formulaires/ctb9.html': 1, 'https://www.canada.ca/en/revenue-agency/services/forms-publications/a-form-publication.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_get-cra-forms': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/individuals/topics/about-your-tax-return/tax-return/completing-a-tax-return/deductions-credits-expenses/deductions-credits-expenses.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_taxes-deductions': 1, 'http://www.canada.ca/en/revenue-agency/services/aboriginal-peoples/information-indians.html#hdng4': 1, 'http://www.canada.ca/taxes-aboriginal-peoples': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/particuliers/segments/deductions-credits-impot-personnes-handicapees/credit-impot-personnes-handicapees.html': 1, 'http://www.cra.gc.ca/myaccount': 1, 'https://www.canada.ca/en/revenue-agency/services/scientific-research-experimental-development-tax-incentive-program.html?utm_campaign=not-applicable&amp;utm_medium=vanity-url&amp;utm_source=canada-ca_taxes-sred': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/entreprises/sujets/tps-tvh-entreprises/secteurs-specifiques/tps-tvh-institutions-financieres-compris-institutions-financieres-designees-particulieres.html': 1, 'https://www.canada.ca/en/revenue-agency/services/tax/canada-pension-plan-cpp-employment-insurance-ei-rulings/cpp-ei-explained/canada-pension-plan-members-religious-sects.html': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/corporate/transparency/access-information-privacy/info-source/personal-information-banks.html#students': 1, 'https://canadianlabour.ca/permanentresidence/': 1, 'www.canada.ca/tps-tvh': 1, 'www.canada.ca/mon-dossier-entreprise-arc': 1, 'www.canada.ca/publications-tps-tvh': 1, 'www.canada.ca/impots-representants': 1, 'www.canada.ca/impotnet-tps-tvh': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/immigrate-canada/rural-northern-immigration-pilot/pr-eligibility.html': 1, 'http://www.cic.gc.ca/english/helpcentre/answer.asp?qnum=125&amp;top=6': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/immigrate-canada/rural-northern-immigration-pilot/proof-funds.html': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/immigrate-canada/express-entry/eligibility/find-national-occupation-code.html': 1, 'http://gcdocs2/otcsdav/nodes/384244698/phac.privacy-vieprivee.aspc%40canada.ca': 1, 'http://canada.ca/cra-forms': 1, 'www.arc.gc.ca': 1, 'cra.gc.ca/forms': 1, 'https://www.canada.ca/fr/agence-revenu/services/impot/particuliers/sujets/tout-votre-declaration-revenus/declaration-revenus/remplir-declaration-revenus/deductions-credits-depenses/ligne-229-autres-depenses-emploi/espace-travail-domicile-depenses.html': 1, 'https://www.cic.gc.ca/english/information/fees/pay.asp': 1, 'https://www.canada.ca/content/dam/ircc/migration/ircc/english/pdf/kits/forms/imm1444e.pdf': 1, 'https://www.fbi.gov/services/cjis/identity-history-summary-checks': 1, 'http://www.canada.ca/calculateur-prestations-enfants-familles': 1, 'http://forums.adobe.com/message/3168183#3168183': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0003-application-canadian-citizenship-minors-under-18-years.html#appendixb': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0003-application-canadian-citizenship-minors-under-18-years.html#appendixa': 1, 'https://www.canada.ca/content/dam/ircc/migration/ircc/english/pdf/kits/citizen/cit0003e-2.pdf': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0003-application-canadian-citizenship-minors-under-18-years.html#appendixc': 1, 'https://www.canada.ca/en/immigration-refugees-citizenship/services/application/application-forms-guides/guide-0003-application-canadian-citizenship-minors-under-18-years.html': 1, 'http://kuler.adobe.com/#themeID/1153604': 1, 'http://kuler.adobe.com/#themeID/1153606': 1, 'http://blogs.adobe.com/formfeed/2008/10/form_variables.html': 1, 'http://blogs.adobe.com/formfeed/': 1, 'http://blogs.adobe.com/formfeed/2011/06/resolvenode-vs-resolvenodes.html': 1, 'http://help.adobe.com/en_US/livecycle/9.0/lcdesigner_scripting_reference.pdf': 1, 'http://partners.adobe.com/public/developer/xml/index_arch.html': 1, 'http://assure.ly/gBUUXB': 1, 'http://assure.ly/h7whb8': 1, 'http://assure.ly/h8ZLr5': 1, 'http://assure.ly/fYCuQ2': 1, 'http://assure.ly/jtTkdO': 1, 'http://assure.ly/gcXx03': 1, 'http://assure.ly/jcTahK': 1, 'http://assure.ly/iitzNJ': 1, 'http://partners.adobe.com/public/developer/en/xml/picture_clause_2.0.pdf': 1, '.#som($template.#subform.designer__stylesheet.a_D_M_Y_Box)': 1, '.#som($template.#subform.designer__stylesheet.a_Head_Level_2_Wide)': 1, '.#som($template.#subform.designer__stylesheet.a_Statement_Violations_Right)': 1, '.#som($template.#subform.designer__stylesheet.a_FooterText)': 1, '.#som($template.#subform.designer__stylesheet.a_Para)': 1, '.#som($template.#subform.designer__stylesheet.a_Para_Bold)': 1, '.#som($template.#subform.designer__stylesheet.a_SingleFieldMax81)': 1, '.#som($template.#subform.designer__stylesheet.a_61_Long_Text_Box_Left)': 1, '.#som($template.#subform.designer__stylesheet.a_Statement_Facts_Continued)': 1, '.#som($template.#subform.designer__stylesheet.a_Para_Wide)': 1, '.#som($template.#subform.designer__stylesheet.a_State_Names)': 1, '.#som($template.#subform.designer__stylesheet.a_AdobeReaderWarning4Lines)': 1, '.#som($template.#subform.designer__stylesheet.a_Head_Level_1)': 1, '.#som($template.#subform.designer__stylesheet.a_TwoLineFieldMax275)': 1, '.#som($template.#subform.designer__stylesheet.a_Warning_Text_Box)': 1, '.#som($template.#subform.designer__stylesheet.a_Head_Level_2.#font)': 1, '.#som($template.#subform.designer__stylesheet.a_Statement_Violations_Left)': 1, '.#som($template.#subform.designer__stylesheet.a_Head_Level_2)': 1, '.#som($template.#subform.designer__stylesheet.a_Text_Field_With_Margins)': 1, '.#som($template.#subform.designer__stylesheet.a_Statement_Facts)': 1, '.#som($template.#subform.designer__stylesheet.a_SingleFieldMax10)': 1, '.#som($template.#subform.designer__stylesheet.a_Head_Reference)': 1, '.#som($template.#subform.designer__stylesheet.a_Button_Text)': 1, '.#som($template.#subform.designer__stylesheet.a_Head_Level_2.#para)': 1, '.#som($template.#subform.designer__stylesheet.a_Head_Level_3)': 1, '.#som($template.#subform.designer__stylesheet.designer__radiobutton_holder.a_RadioButton)': 1, '.#som($template.#subform.designer__stylesheet.a_Para_Bold.#para)': 1, '.#som($template.#subform.designer__stylesheet.a_Para_Hanging)': 1, '.#som($template.#subform.designer__stylesheet.a_Section_I_List)': 1, '.#som($template.#subform.designer__stylesheet.a_Para_Bold_Wide)': 1, '.#som($template.#subform.designer__stylesheet.a_Address_Field_No_Limit_9_Lines)': 1, '.#som($template.#subform.designer__stylesheet.a_Date_Field)': 1, '.#som($template.#subform.designer__stylesheet.a_Checkbox)': 1, '.#som($template.#subform.designer__stylesheet.a_61_Long_Text_Box_Right)': 1, '.#som($template.#subform.designer__stylesheet.a_Header_Right_Num_Blue)': 1, '.#som($template.#subform.designer__stylesheet.a_Para_Hanging.#font)': 1, '.#som($template.#subform.designer__stylesheet.a_RadioButtonGroup)': 1, '.#som($template.#subform.designer__stylesheet.a_Date_Example)': 1, '.#som($template.#subform.designer__stylesheet.a_Header_Left_Title_Blue)': 1, '.#som($template.#subform.designer__stylesheet.a_Para.#font)': 1, '.#som($template.#subform.designer__stylesheet.a_Section_I_List_Num)': 1, '.#som($template.#subform.designer__stylesheet.a_Para_Hanging.#para)': 1, '.#som($template.#subform.designer__stylesheet.a_Para.#para)': 1, '.#som($template.#subform.designer__stylesheet.a_Para_Hanging_Wide)': 1, '.#som($template.#subform.designer__stylesheet.a_Head_Level_1_Right)': 1, '.#som($template.#subform.designer__stylesheet.a_Address_Field_No_Limit_6_Lines)': 1, '.#som($template.#subform.designer__stylesheet.a_Para_Bold.#font)': 1, 'https://www.gov.uk/government/publications/personal-information-use-in-borders-immigration-and-citizenship': 1, 'www.gov.uk/government/organisations/home-office': 1, 'mailto:DomesticViolence2@homeoffice.gov.uk': 1
```

Reply:
there are few relative paths but only for some images but we get only images with their data somewhere in the pdf:
https://github.com/mozilla/pdf.js/blob/580bfad6288ca17b90db2dea90914103394d4c1d/src/core/xfa/template.js#L3354

Reply:
There are also some `mailto:`, one `#reportTop` and some relative URLs to PDFs (such as `../pdf/quickstarts/rbm signer.pdf`).